crates/gpui/src/platform/linux/platform.rs

```diff
@@ -594,6 +597,46 @@ impl Modifiers {
     }
 }
 
+// From https://github.com/wez/wezterm/blob/95581d8697f3749f84ccb1402ac94ea6582b227f/window/src/os/wayland/window.rs#L468
+pub(super) fn read_pipe_with_timeout(mut file: FileDescriptor) -> Result<String> {
```

Comment:
This was inspired by https://github.com/zed-industries/zed/pull/8365, which originally took the code from wezterm, which is MIT licensed.